kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -208,6 +208,31 @@ protected Tuple2<Protocol, Metadata> loadTableProtocolAndMetadata(
       return new Tuple2<>(snapshotHint.get().getProtocol(), snapshotHint.get().getMetadata());
     }
 
+    // Compute the lower bound for the CRC search
+    // If the snapshot hint is present, we can use it as the lower bound for the CRC search.
+    // TODO: this can be further improved to make the lower bound until the checkpoint version
+    Optional<Long> crcSearchLowerBound = snapshotHint.map(SnapshotHint::getVersion);
+    Optional<VersionStats> versionStatsOpt =
+        ChecksumReader.getVersionStats(
+            engine, logSegment.logPath, snapshotVersion, crcSearchLowerBound);
+    if (versionStatsOpt.isPresent()) {
+      // We found the protocol and metadata for the version we are looking for
+      VersionStats versionStats = versionStatsOpt.get();
+      if (versionStats.getVersion() == snapshotVersion) {
+        return new Tuple2<>(versionStats.getProtocol(), versionStats.getMetadata());
+      }
+      // We found the protocol and metadata in a version older than the one we are looking
```

Comment:
Let's make this a bit clearer. Please mention that the info from the CRC is newer than the snapshot hint, that's why we update it

Reply:
I wondering if following is the valid case -- for time travel, we want to look at the CRC earlier than the snapshot hint, if snapshot hint points to the latest version.

Reply:
That's a great observation! Why don't we do this: in the constructor of LogReplay, let's check if the snapshotHint's version is > our target snapshotVersion. If it is, then set the snapshotHint to Optional.empty

Reply:
`<<< LogReplay doesn't own snapshotHint, but just uses it for a fast path of loading P&M >>>`

I agree with this statement. What does this have to do with `<<< let's check if the snapshotHint's version is > our target snapshotVersion. If it is, then set the snapshotHint to Optional.empty >>>` ?

If we set the variable `snapshotHint` to `Optional.empty`, this doesn't impact the original value of snapshotHint from the caller of the method.